adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceBannerAd.m

```diff
@@ -0,0 +1,135 @@
+@property(nonatomic, copy) NSString *instanceID;
+
+/// Holds the state of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceState;
+
+@property (nonatomic, strong) ISDemandOnlyBannerView *iSDemandOnlyBannerView;
+
+@end
+
+@implementation GADMAdapterIronSourceBannerAd
+
+- (instancetype)initWithGADMediationBannerConfiguration:(GADMediationBannerAdConfiguration *)adConfiguration
+                                      completionHandler:(GADMediationBannerLoadCompletionHandler)completionHandler {
+    self = [super init];
+    if (self) {
+        _adLoadCompletionHandler = completionHandler;
+        _adConfiguration = adConfiguration;
+        // Default instance ID
+        self.instanceID = GADMIronSourceDefaultInstanceId;
+        // Default instance state
+        self.instanceState = GADMAdapterIronSourceInstanceStateStart;
+    }
+    
+    return self;
+}
+
+- (void)renderBannerForAdConfig:(nonnull GADMediationBannerAdConfiguration *)adConfig
+              completionHandler:(nonnull GADMediationBannerLoadCompletionHandler)handler {
+    _adConfiguration = adConfig;
+    NSDictionary *credentials = [_adConfiguration.credentials settings];
+    
+    /* Parse application key */
+    NSString *applicationKey = @"";
+    if (credentials[GADMAdapterIronSourceAppKey]) {
+        applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    }
+    
+    if ([GADMAdapterIronSourceUtils isEmpty:applicationKey]) {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorInvalidServerParameters,
+                                                                          @"'appKey' parameter is missing. Make sure that appKey' server parameter is added.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if (credentials[GADMAdapterIronSourceInstanceId]) {
+        self.instanceID = credentials[GADMAdapterIronSourceInstanceId];
+    }
+    
+    [[ISMediationManager sharedManager]
+     initIronSourceSDKWithAppKey:applicationKey
+     forAdUnits:[NSSet setWithObject:IS_BANNER]];
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        ISMediationManager *sharedManager = [ISMediationManager sharedManager];
+        [sharedManager initIronSourceSDKWithAppKey:applicationKey
+                                        forAdUnits:[NSSet setWithObject:IS_BANNER]];
+        GADAdSize adSize = adConfig.adSize;
+        ISBannerSize *ISBannerSize = [GADMAdapterIronSourceUtils ironSourceAdSizeFromRequestedSize:adSize];
+        [[ISMediationManager sharedManager]loadBannerAdWithDelegate:self viewController:self instanceID:_instanceID bannerSize:ISBannerSize];
+    });
+}
+
+- (void)bannerDidFailToLoadWithError:(NSError *)error
+                          instanceId:(NSString *)instanceId {
+    [_adEventDelegate didFailToPresentWithError:error];
```

Comment:
This is a presentation error.
To forward a load error, pass an error object to `adLoadCompletionHandler` as shown below:

`_adLoadCompletionHandler(nil, error);`

Reply:
Fixed